### ISSUE TITLE
Users need to view good quality error messages so that they know what to enter in each field

### ISSUE THREAD

**Issue by EUzkuraityte** [alphagov/verify-local-patterns#557]
Hi @lizziebruce @sanjaypoyzer this is not urgent, but we'll need he error messages incorporated into our prototypes at some point.

This can be looked into at some point in the future and if you'd prefer to keep the prototype error message free, it'd still be good to issue some guidance to the councils to advice what a good error message looks like. Let me know your thoughts on this!

**Comment by lizziebruce** [comment]
@EUzkuraityte @sanjaypoyzer 

Error messages – these are for all the pages where users type things in. They would show when invalid data entered in the relevant field. May need some dev time to implement these?

**Enter card details page**
> Check the card details you’ve entered are correct.
> Check the billing address you’ve entered is correct.
> Enter a valid email address.

**Enter card details – payment failed**
> It hasn’t been possible to process your payment. Use a different payment card or payment method. 

**Pay by Direct Debit**
> Check the name you’ve entered for your bank account is correct.
> Check the sort code you’ve entered is correct.
> Check the account number you’ve entered is correct.
> Check the billing address you’ve entered is correct.
> Enter a valid email address.

**Pay by Direct Debit – payment failed**
> It hasn’t been possible to set up a Direct Debit with this bank account. Use a different bank account or payment method. 

**Check boundary**
> Enter a valid UK postcode.

**Enter vehicle reg**
> Check the vehicle registration number(s) you’ve entered are correct.

**Contact reference**
> Enter a valid email address.
> Enter a valid mobile phone number.

**Contact notifications**
> Enter a valid email address.
> Enter a valid mobile phone number.
> Check the address you’ve entered is correct.

**Comment by EUzkuraityte** [comment]
@lizziebruce many thanks for putting this together, looks good! I'll wait for @sanjaypoyzer to let us know if dev time is needed.. Though chances of getting any are slim anyway. We can still pass this on to LAs and tell them that when they build the product these are the messages that should be shown. Additionally, are you able to put together a couple of sentences to make it into guidance to the councils to advice what a good error message looks like and how to come up with them?
@ingasiler any chance you could save this somewhere on the shared drive so we can pass this on to LAs for implementation as well?

**Comment by lizziebruce** [comment]
@EUzkuraityte @ingasiler 

Guidance or error messages

> If the error is likely to be caused by the user mistyping something, like a number or text, use the phrase "Check the xxx you've entered is correct.
> If the error is likely to be caused by the user typing something in the wrong format, like a mobile number without the right number of digits or an email without an @ or a domain name, use the phrase "Enter a valid xxx". However best practice is to allow for users inputting data in a variety of formats, for example the postcode field should allow users to enter a postcode with or without a space. 

Does the above sound okay in relation to the prototype design @sanjaypoyzer ?

**Comment by suryanaga** [comment]
Yep, the error messages look fine -- Me or a front end dev also needs to have a think about what the actual validation rules for each form field are. Probably more a thing for beta than alpha though.
